src/background/availabilityService.js

```diff
@@ -0,0 +1,99 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @ts-check
+import { Component } from "./component.js";
+import { PropertyType } from "../shared/ipc.js";
+
+import { property, WritableProperty } from "../shared/property.js";
+import { VPNController } from "./vpncontroller/vpncontroller.js";
+
+/**
+ *
+ * AvailablityService reports if a user will be able to
+ * create a subscription in at the current location.
+ */
+
+export class AvailablityService extends Component {
+  // Gets exposed to UI
+  static properties = {
+    isAvailable: PropertyType.Bindable,
+    check: PropertyType.Function,
+    waitlistURL: PropertyType.Bindable,
+    ignore: PropertyType.Function,
+  };
+
+  /** @type {WritableProperty<String>} */
+  // Availablity status:
+  // Valid Strings: "pending, available, unavailable, ignored"
+  isAvailable = property("pending");
+  /** @type {WritableProperty<String>} */
+  waitlistURL = property("");
+
+  async ignore() {
+    this.isAvailable.value = "ignored";
+  }
+
+  async check() {
+    return await fetch("https://www.mozilla.org/products/vpn/", {
```

Comment:
Nit: A bit perplexed by the async/await and chained .thens but I don't need to be picky about it.